verification/cbmc/proofs/derive_data_key/derive_data_key_harness.c

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"). You may not
+ * use
+ * this file except in compliance with the License. A copy of the License is
+ * located at
+ *
+ *     http://aws.amazon.com/apache2.0/
+ *
+ * or in the "license" file accompanying this file. This file is distributed on
+ * an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or
+ * implied. See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <aws/cryptosdk/private/cipher.h>
+#include <aws/cryptosdk/private/header.h>
+#include <aws/cryptosdk/private/hkdf.h>
+#include <aws/cryptosdk/private/session.h>
+#include <aws/cryptosdk/session.h>
+#include <make_common_data_structures.h>
+
+void derive_data_key_harness() {
+    /* Nondet input */
+    struct aws_cryptosdk_session *session         = ensure_session_attempt_allocation(MAX_STRING_LEN);
+    struct aws_cryptosdk_dec_materials *materials = ensure_dec_materials_attempt_allocation();
+
+    /* Assumptions */
+    __CPROVER_assume(session != NULL);
+    __CPROVER_assume(aws_cryptosdk_commitment_policy_is_valid(session->commitment_policy));
+
+    __CPROVER_assume(session->alg_props != NULL);
+    __CPROVER_assume(aws_cryptosdk_alg_properties_is_valid(session->alg_props));
+    __CPROVER_assume(session->alg_props->commitment_len <= sizeof(session->key_commitment_arr));
+
+    __CPROVER_assume(aws_byte_buf_is_bounded(&session->header.message_id, MAX_BUFFER_SIZE));
+    ensure_byte_buf_has_allocated_buffer_member(&session->header.message_id);
+    __CPROVER_assume(aws_byte_buf_is_valid(&session->header.message_id));
+
+    __CPROVER_assume(aws_byte_buf_is_bounded(&session->header.alg_suite_data, MAX_BUFFER_SIZE));
+    ensure_byte_buf_has_allocated_buffer_member(&session->header.alg_suite_data);
+    __CPROVER_assume(aws_byte_buf_is_valid(&session->header.alg_suite_data));
```

Comment:
Replace that by
```c
__CPROVER_assume(aws_cryptosdk_session_is_valid(session));
```

Reply:
We do have a `session_setup` function which boths performs allocation and common assumptions.